renderSpan: add more time intervals

j-s-n opened the pull request:
## Summary
Added years, months, and days to renderSpan
Estimated years to 365 days and months to 30 days.

## Testing Plan
Lock and unlock a wallet with flags for various time intervals.
Confirmed it returns the highest 2 time units (year & month, month & day, etc.)

## Documentation
N/A

## Breaking Change
N/A

NullSoldier commented on ironfish/src/utils/time.ts:
```diff
@@ -58,6 +68,29 @@ const renderSpan = (
 
   const parts = []
   let magnitude = 0
+  // Do we want to include years? Is it ok to estimate at 365 days?
```
Yes I think it's ok to include years and to estimate at 365 days

j-s-n replied:
Update year calculation to use 365 days.

NullSoldier commented on ironfish/src/utils/time.ts:
```diff
@@ -58,6 +68,29 @@ const renderSpan = (
+    magnitude = Math.max(magnitude, 7)
+  }
+
+  // Do we want to include months? Is it ok to estimate at 30 days?
```
Yes, I think it's ok to include months and to estimate at 30 days a month.

j-s-n replied:
Kept month calculation at 30 days and updated month abbreviation to 'M'. capital M is commonly used in other libraries (e.g. y M d h m s ms)

NullSoldier commented:
I think we should add a few tests in the test file for these new cases.

j-s-n commented:
> I think we should add a few tests in the test file for these new cases.

added tests for renderSpan and renderEstimate. Updated renderEstimate to return largest 3 intervals (y, M, d) for longer periods and renderSpan outputs largest 2 intervals (e.g. y, M / M, d / d, h)

NullSoldier commented:
Ok the PR looks good. Assuming the tests pass we should merge this in.

NullSoldier commented on ironfish/src/utils/time.ts:
```diff
@@ -58,6 +67,26 @@ const renderSpan = (
 
   const parts = []
   let magnitude = 0
```
```suggestion
  let magnitude = 0

```
